### PR TITLE
Longest subsequence java

### LINKED ISSUE #1554: Longest_Increasing_Subsequence code in Java
@adishjain @ayushin78 @nj4710 
As a GSSoC participant, I would like to work  on this, please assign this task  to me.

### PR DESCRIPTION
**Pull request by BhavishayGulati**
Fixes #1554

#### Checklist:
- [x] 4 space indentation.
- [x] Coding conventions are followed.
- [x] Input is taken dynamically.
- [x] Sample Input / Output is added at the end of file.
- [x] Logic Documentation (Comments).
- [x] File names are correct.

#### Changes proposed in this pull request:
- code added for the algorithm for the longest increasing subsequence

#### Languages Used:
- Java

#### Files Added:
- longest_increasing_subsequence_java

### CODE REVIEW

File: Longest_Increasing_subsequence_java/longest_increasing_subsequence_java.java [hunk 1]
```diff
@@ -0,0 +1,48 @@
+package com.company;
+
+
```
Comment (SakshiShreya) [inline Longest_Increasing_subsequence_java/longest_increasing_subsequence_java.java]: Please remove the package and the extra new-lines.
Also add documentation at the beginning of the file.

File: Longest_Increasing_subsequence_java/longest_increasing_subsequence_java.java [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+            return 0;
+        }
+
+        int[] cache = new int[a.length];
+        Arrays.fill(cache, 1);
+
+        for (int i = 1; i < a.length; i++) {
+            for (int j = 0; j < i; j++) {
+                if (a[i] > a[j]) {
+                    cache[i] = Math.max(cache[i], cache[j] + 1); // checking for the longest increasing subsequence
```
Comment (SakshiShreya) [inline Longest_Increasing_subsequence_java/longest_increasing_subsequence_java.java]: Move the comment to previous or next line. One line can't contain more than 80 characters including indentation.

File: Longest_Increasing_subsequence_java/longest_increasing_subsequence_java.java [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+                }
+            }
+        }
+
+        return Arrays.stream(cache).max().getAsInt();       // returning the size of the longest increasing subsequence
```
Comment (SakshiShreya) [inline Longest_Increasing_subsequence_java/longest_increasing_subsequence_java.java]: Maintain character limit here also.

File: Longest_Increasing_subsequence_java/longest_increasing_subsequence_java.java [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+    }
+
+
```
Comment (SakshiShreya) [inline Longest_Increasing_subsequence_java/longest_increasing_subsequence_java.java]: There should be only one new-line between two functions.

**Comment by jainaman224** [comment]
@BhavishayGulati Are you working on it?

**Comment by somya-kapoor** [comment]
@BhavishayGulati Are you working on it?

**Comment by somya-kapoor** [comment]
@BhavishayGulati Due to no activity, I am closing this PR.

**Pull request closed without merging**